.github/workflows/clang-tidy.yml

```diff
@@ -0,0 +1,151 @@
+---
+name: 'clang-tidy'
+
+on:
+  pull_request:
+    branches: [master]
+
+permissions:
+  contents: read
+
+jobs:
+  clang-tidy-auth:
+    name: auth clang-tidy
+    runs-on: ubuntu-20.04
+    env:
+      UNIT_TESTS: yes
+      SANITIZERS:
+    steps:
+      - uses: PowerDNS/pdns/set-ubuntu-mirror@meta
+      - uses: actions/checkout@v3.1.0
+        with:
+          fetch-depth: 2
+      - name: get timestamp for cache
+        id: get-stamp
+        shell: bash
+        run: |
+          echo "stamp=$(/bin/date +%s)" >> "$GITHUB_OUTPUT"
+      - name: let GitHub cache our ccache data
+        uses: actions/cache@v3.2.5
+        with:
+          path: ~/.ccache
+          key: auth-ccache-${{ steps.get-stamp.outputs.stamp }}
+          restore-keys: auth-ccache-
```

Comment:
I wonder if we need a different cache key than the one we used in other workflows, given than we build with different options (no sanitizers)?

Reply:
Isn't this already a difference cache key? The ones from `build-and-test-all.yml` have the sanitizer name in them which doesn't exist here.

Reply:
They do for dnsdist and the recursor, but not for the authoritative, unless I'm mistaken?

Reply:
Ah, correct.